certora/specs/Reverts.spec

```diff
@@ -176,21 +178,29 @@ rule setMaxRateRevertCondition(env e, uint256 newMaxRate) {
     assert lastReverted <=> e.msg.value != 0 || !callerIsAllocator || newMaxRate > maxMaxRate;
 }
 
-rule transferInputValidation(env e, address to, uint256 shares) {
+rule transferRevertCondition(env e, address to, uint256 shares) {
     bool toIsZeroAddress = to == 0;
     bool callerCanSendShares = canSendShares(e.msg.sender);
     bool toCanReceiveShares = canReceiveShares(to);
 
+    require shares + balanceOf(to) <= MAX_UINT256(); // To avoid overflow in the balanceOf check.
+    require shares <= balanceOf(e.msg.sender); // To avoid underflow in the balanceOf check.
```

Comment:
let's write these as elements of the or in the assert line (to have all the requirements at he same place)

Reply:
more than having them all at the same place actually, here we are not verifying the equivalence (see [this bug](https://github.com/morpho-org/vault-v2/pull/873/changes#r2653486098))

Reply:
good point! 
agree that using require is not logically equivalent to declaring a boolean and adding it as a condition in the assert statement. The proofs fail with the proposed changes, i.e., !balanceWontOverflow is true but lastReverted is false. This happens when `to == e.msg.sender`.   So the condition should be modified into `to!= e.msg.sender => shares + balanceOf(to) <= MAX_UINT256()`